apps/server/src/modules/tool/external-tool/mapper/external-tool-datasheet.mapper.ts

```diff
@@ -0,0 +1,170 @@
+import { Configuration } from '@hpi-schul-cloud/commons/lib';
+import {
+	ExternalTool,
+	ExternalToolDatasheetTemplateData,
+	ExternalToolParameterDatasheetTemplateData,
+	ExternalToolParameterDatasheetTemplateProperty,
+} from '../domain';
+import { CustomParameterScope, CustomParameterType, ToolConfigType, ToolContextType } from '../../common/enum';
+import { CustomParameter } from '../../common/domain';
+
+export class ExternalToolDatasheetMapper {
+	public static mapToExternalToolDatasheetTemplateData(
+		externalTool: ExternalTool,
+		firstName: string,
+		lastname: string
+	): ExternalToolDatasheetTemplateData {
+		const externalToolData: ExternalToolDatasheetTemplateData = new ExternalToolDatasheetTemplateData({
+			createdAt: new Date().toLocaleDateString('de-DE'),
+			creatorName: `${firstName} ${lastname}`,
+			instance: ExternalToolDatasheetMapper.mapToInstanceName(),
+			toolName: externalTool.name,
+			toolUrl: externalTool.config.baseUrl,
+			isDeactivated: externalTool.isDeactivated ? 'Das Tool ist deaktiviert' : undefined,
+			restrictToContexts: externalTool.restrictToContexts
+				? ExternalToolDatasheetMapper.mapToLimitedContexts(externalTool)
+				: undefined,
+			toolType: ExternalToolDatasheetMapper.mapToToolType(externalTool),
+		});
+
+		if (externalTool.parameters) {
+			externalToolData.parameters = ExternalToolDatasheetMapper.mapToParameterDataList(externalTool);
+		}
+
+		if (ExternalTool.isOauth2Config(externalTool.config)) {
+			if (externalTool.config.skipConsent) {
+				externalToolData.skipConsent = 'ja';
+			}
+		}
+
+		if (ExternalTool.isLti11Config(externalTool.config)) {
+			externalToolData.messageType = externalTool.config.lti_message_type.toString();
+			externalToolData.privacy = externalTool.config.privacy_permission.toString();
+		}
+
+		return externalToolData;
+	}
+
+	private static mapToInstanceName(): string {
+		if (Configuration.get('SC_THEME') === 'n21') {
+			return 'Niedersächsische Bildungscloud';
+		}
+
+		if (Configuration.get('SC_THEME') === 'brb') {
+			return 'Schul-Cloud Brandenburg';
+		}
+
+		if (Configuration.get('SC_THEME') === 'thr') {
+			return 'Thüringer Schulcloud';
+		}
+
+		if (Configuration.get('SC_THEME') === 'default') {
+			return 'dBildungscloud';
+		}
+
+		return 'unbekannt';
+	}
+
+	private static mapToLimitedContexts(externalTool: ExternalTool): string {
+		const restrictToContexts: string[] = [];
+		if (externalTool.restrictToContexts?.includes(ToolContextType.COURSE)) {
+			restrictToContexts.push('Kurs');
+		}
+		if (externalTool.restrictToContexts?.includes(ToolContextType.BOARD_ELEMENT)) {
+			restrictToContexts.push('Kurs-Board');
+		}
+
+		const restrictToContextsString = restrictToContexts.join(', ');
+
+		return restrictToContextsString;
+	}
+
+	private static mapToToolType(externalTool: ExternalTool): string {
+		let toolType: string = externalTool.config.type;
+		if (externalTool.config.type === ToolConfigType.OAUTH2) {
```

Comment:
toolType is possably not translated (in case of basic). use a different variable for the return value or use a switch.

Reply:
our BA did not wanted this to be translated, because in his eyes basic is just basic 🤷‍♂️ I adopted the switch style though :)

Reply:
then you should still add a mapping because it could be that we rename the internal name of the type and then it does not match the expectation anymore. An he probably wants to have "Basic" as the name not "basic".

Reply:
Though this sounds absolutely logical, i am not sure about what our BA wants 😅 I do think, that he does not care to much about it and it will work with 'Basic' as well ^^